crates/sui-node/src/lib.rs

```diff
@@ -139,6 +139,7 @@ impl SuiNode {
 
         let secret = Arc::pin(config.protocol_key_pair().copy());
         let genesis_committee = genesis.committee()?;
+        let reference_gas_price = genesis.sui_system_object().reference_gas_price;
```

Comment:
This may not work since when the node starts up, the state of the system state may not be the same as the epoch of the node. I think you will need to persist the gas price to the db.
I would recommend adding this field to `EpochStartConfiguration` and then cache it in memory

Reply:
Specifically, https://github.com/MystenLabs/sui/pull/8364 just did it for you

Reply:
Yes, we should definitely use EpochStartConfiguration here. Lmk if you want me to put a PR to pass entire SuiSystemState to EpochStartConfiguration (or feel free to just do it here if you want)

Reply:
Made https://github.com/MystenLabs/sui/pull/8366 to pass entire sui system state there

Reply:
let me see if I am following here.
We are running here at any point in time, say, because a node goes down for whatever reason.
It starts again and it needs the RGP related to the epoch it is starting in.
Here I am always passing the genesis one and that is wrong.
But if I follow the code we should be loading the DB which contains the snapshot of where the node was at, and we should have stored the RGP in there so all is good.
Is that a good understanding on my part?

Reply:
Yes, EpochStartConfiguration would have a snapshot of RGP in the beginning of current epoch. (I have not yet read PR fully, but you also obviously need to update this on epoch change)